packages/runtime/src/state/react-builder-preview.ts

```diff
@@ -197,6 +197,36 @@ function getElementSize(element: HTMLElement): Size {
   }
 }
 
+function handleFocusEvent(): ThunkAction<() => void, unknown, unknown, Action> {
```

Comment:
Why make this a thunk if it doesn't rely on `dispatch` or `getState`?

Reply:
Also, the naming is a bit confusing. This is not an event handler, it's more of an effect since it returns a cleanup function.

Reply:
Some suggestions: `lockDocumentScroll`, `lockDocumentScrollEffect`, `setUpFocusListeners`, `setUpFocusListenersEffect`.

Reply:
Another one, `startHandlingFocusEvents`.